### PR TITLE
chore: bump deps

### PR DESCRIPTION
**Pull request by DmitriyMV**
Bump:
- cilium/cilium-cli to v0.15.2
- github.com/Azure/azure-sdk-for-go/sdk/azcore to v1.7.0
- github.com/Azure/azure-sdk-for-go/sdk/storage/azblob to v1.1.0
- github.com/aws/aws-sdk-go to v1.44.300
- github.com/beevik/ntp to v1.2.0
- github.com/docker/docker to v24.0.4+incompatible
- github.com/gomarkdown/markdown to v0.0.0-20230711084535-11b03c0ae6d6
- github.com/hetznercloud/hcloud-go to v1.48.0
- github.com/iancoleman/orderedmap to v0.3.0
- github.com/jsimonetti/rtnetlink to v1.3.4
- github.com/siderolabs/go-debug to v0.2.3
- golang.org/x/net to v0.12.0
- golang.org/x/tools to v0.11.0
- google.golang.org/genproto/googleapis/rpc to v0.0.0-20230711160842-782d3b101e98
- google.golang.org/grpc to v1.56.2
- google.golang.org/protobuf to v1.31.0

### CODE REVIEW

**Comment by DmitriyMV** [comment]
Since cilium integration tests started failing with [nil pointer panic](https://ci.dev.talos-systems.io/siderolabs/talos/35120/1/5) I want to see if its because our changes in #7474 or because another new bug in cilium-cli.

File: go.mod [hunk 1]
```diff
@@ -309,4 +309,7 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
-require github.com/google/gnostic-models v0.6.8 // indirect
+require (
```
Comment (frezbo) [inline go.mod:312]: why this in a separate require block, even for indirect deps :thinking:

**Review by frezbo — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (DmitriyMV) [reply go.mod]: `go mod tidy` or Goland inconsistency I think?

Comment (DmitriyMV) [reply go.mod]: Fixed

**Review by frezbo — APPROVED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**